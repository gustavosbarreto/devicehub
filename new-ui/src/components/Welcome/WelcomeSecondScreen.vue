<template>
  <p class="ml-4 pt-4 text-subtitle-2">
    In order to register a device on ShellHub, you need to install ShellHub
    agent onto it.
  </p>
  <p class="ml-4 pt-4 text-subtitle-2">
    The easiest way to install ShellHub agent is with our automatic one-line
    installation script, which works with all Linux distributions that have
    Docker installed and properly set up.
  </p>
  <div class="mt-4 ml-4 mr-4">
    <p class="ml-2 pt-4 text-subtitle-2 text-bold">
      Run the following command on your device:
    </p>

    <v-text-field
      :model-value="command"
      @click:append="copyCommand"
      class="code mt-1"
      variant="outlined"
      append-icon="mdi-content-copy"
      readonly
      active
      data-test="command-field"
      density="compact"
    />

    <v-divider />

    <p class="text-caption mb-0 mt-1">
      Check the
      <a
        :href="'https://docs.shellhub.io/user-guides/devices/adding'"
        target="_blank"
        >documentation</a
      >
      for more information and alternative install methods.
    </p>
  </div>
</template>

<script lang="ts">
import { INotificationsCopy } from "../../interfaces/INotifications";
import { useStore } from "../../store";
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    command: {
      type: String,
      required: true,
    },
  },

  setup(props) {
    const store = useStore();
    const copyCommand = () => {
      navigator.clipboard.writeText(props.command);
      store.dispatch("snackbar/showSnackbarCopy", INotificationsCopy.tenantId);
    };

    return {
      copyCommand,
    };
  },
});
</script>
