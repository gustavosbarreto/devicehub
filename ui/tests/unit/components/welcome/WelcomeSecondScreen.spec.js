import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import WelcomeSecondScreen from '@/components/welcome/WelcomeSecondScreen';

describe('WelcomeSecondScreen', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const command = 'curl "http://localhost/install.sh?tenant_id=a582b47a42e" | sh';

  const store = new Vuex.Store({
    namespaced: true,
    state: {
    },
    getters: {
    },
    actions: {
      'snackbar/showSnackbarCopy': () => {
      },
    },
  });

  beforeEach(() => {
    wrapper = shallowMount(WelcomeSecondScreen, {
      store,
      localVue,
      stubs: ['fragment'],
      propsData: { command },
    });
  });

  it('Is a Vue instance', () => {
    expect(wrapper).toBeTruthy();
  });
  it('Renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Receive data in props', () => {
    expect(wrapper.vm.command).toEqual(command);
  });
  it('Renders the template with data', () => {
    const commandText = wrapper.find('[data-test="command-text"]');
    commandText.element.value = command;
    expect(commandText.element.value).toEqual(command);
  });
});
