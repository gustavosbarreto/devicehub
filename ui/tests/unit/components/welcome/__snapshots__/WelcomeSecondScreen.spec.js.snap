// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSecondScreen Renders the component 1`] = `
<fragment-stub>
  <p class="ml-4 pt-4">
    In order to register a device on ShellHub, you need to install ShellHub agent onto it.
  </p>
  <p class="mt-4 ml-4 mr-4">
    The easiest way to install ShellHub agent is with our automatic one-line installation script,
    which works with all Linux distributions that have Docker installed and properly set up.
  </p>
  <div class="mt-4 ml-4 mr-4"><strong>Run the following command on your device:</strong>
    <v-text-field-stub errorcount="1" errormessages="" messages="" readonly="true" rules="" successmessages="" value="curl -sSf &quot;http://localhost/install.sh?tenant_id=a582b47a42e&quot; | sh" appendicon="mdi-content-copy" backgroundcolor="" dense="true" loaderheight="2" clearicon="$clear" outlined="true" type="text" fill-width="" data-test="command-text" class="code"></v-text-field-stub>
    <v-divider-stub></v-divider-stub>
    <p class="caption mb-0">
      Check the <a href="https://shellhub-io.github.io/guides/registering-device/" target="_blank">documentation</a>
      for more information and alternative install methods.
    </p>
  </div>
</fragment-stub>
`;
